pkg/policies/action/action.go

```diff
@@ -0,0 +1,757 @@
+	// not be matched by this selector.
+	Exclude []*RepoSelector `json:"exclude"`
+}
+
+// ActionSelector specifies a selection of Actions
+type ActionSelector struct {
+	// Name is the Action name in glob format
+	Name string `json:"name"`
+
+	// Version is a semver condition or commit ref
+	// Default "" targets any version
+	Version string `json:"version"`
+}
+
+type details struct {
+	FailedRules []*Rule
+}
+
+type workflowMetadata struct {
+	filename string
+	workflow *actionlint.Workflow
+}
+
+type actionMetadata struct {
+	name             string
+	version          string
+	workflowFilename string
+	workflowName     string
+}
+
+var configFetchConfig func(context.Context, *github.Client, string, string, string, config.ConfigLevel, interface{}) error
+
+var listWorkflows func(ctx context.Context, c *github.Client, owner, repo string, on []string) ([]*workflowMetadata, error)
+var listLanguages func(ctx context.Context, c *github.Client, owner, repo string) (map[string]int, error)
+var listWorkflowRunsByFilename func(ctx context.Context, c *github.Client, owner, repo string, workflowFilename string) ([]*github.WorkflowRun, error)
+var getLatestCommitHash func(ctx context.Context, c *github.Client, owner, repo string) (string, error)
+var listReleases func(ctx context.Context, c *github.Client, owner, repo string) ([]*github.RepositoryRelease, error)
+
+func init() {
+	configFetchConfig = config.FetchConfig
+	listWorkflows = githubListWorkflows
+	listLanguages = githubListLanguages
+	listWorkflowRunsByFilename = githubListWorkflowRunsByFilename
+	getLatestCommitHash = githubGetLatestCommitHash
+	listReleases = githubListReleases
+}
+
+// sortableRules is a sortable list of *Rule
+type sortableRules []*Rule
+
+// Action is the Action Use policy object, implements policydef.Policy.
+type Action bool
+
+// NewAction returns a new Action Use policy.
+func NewAction() policydef.Policy {
+	var a Action
+	return a
+}
+
+// Name returns the name of this policy, implementing policydef.Policy.Name()
+func (a Action) Name() string {
+	return polName
+}
+
+// Check performs the policy check for Action Use policy based on the
+// configuration stored in the org, implementing policydef.Policy.Check()
+func (a Action) Check(ctx context.Context, c *github.Client, owner,
+	repo string) (*policydef.Result, error) {
+	oc := getConfig(ctx, c, owner, repo)
+	enabled := oc.Groups != nil
+	log.Info().
+		Str("org", owner).
+		Str("repo", repo).
+		Str("area", polName).
+		Bool("enabled", enabled).
+		Msg("Check repo enabled")
+	if !enabled {
+		// Don't run this policy if no rules exist.
+		return &policydef.Result{
+			Enabled:    enabled,
+			Pass:       true,
+			NotifyText: "Disabled",
+			Details:    details{},
+		}, nil
+	}
+	// Get workflows.
+	// Workflows should have push and pull_request listed as trigger events
+	// in order to qualify.
+	wfs, err := listWorkflows(ctx, c, owner, repo, []string{"push", "pull_request"})
```

Comment:
I don't remember seeing this in the design. Not sure this is the complete list. Likely for deny, we'd want to check all workflows.